packages/cli/src/ActiveExecutions.ts

```diff
@@ -75,6 +83,17 @@ export class ActiveExecutions {
 			await this.executionRepository.updateExistingExecution(executionId, execution);
 		}
 
+		return await this.enqueue(executionId, executionData, executionStatus);
+	}
+
+	async enqueue(
+		executionId: string,
+		executionData: IWorkflowExecutionDataProcess,
+		executionStatus: ExecutionStatus,
+	) {
+		// Wait here in-case execution concurrency limit is reached
+		await this.getQueue(executionData.executionMode).enqueue(executionId);
```

Comment:
I'd've expected all this to be encapsulated? As in, a `ConcurrencyService` that contains the two queues and is smart enough to identify manual and production executions without being told what kind they are, e.g. `this.concurrencyService.enqueue(execution)`